crates/sui-graphql-rpc/src/consistency.rs

```diff
@@ -166,6 +167,7 @@ pub(crate) fn build_objects_query(
                 newer
             );
             history_objs = filter!(history_objs, "newer.object_version IS NULL");
+            history_objs = filter!(history_objs, "object_status = 0");
```

Comment:
@wlmyng do you think we will need an index
>  (object_id, object_version, checkpoint_sequence_number, object_status)

instead of

>  (object_id, object_version, checkpoint_sequence_number)

Reply:
I think we would benefit more from having the existing indexes replaced with partial indexes, except for the index on `(object_id, object_version, checkpoint_sequence_number)`. This is because if we apply the filter on `object_status`, basically every index that supports a possible filter will also need to account for `object_status` in some way. Since `object_status` consists of 1 of 2 values, a partial index will do the job and comes with the added benefit of not indexing `WrappedOrDeleted` entries as well

The reason why I think we don't want a partial index on `(object_id, object_version, checkpoint_sequence_number)` is because of `objects_snapshot`:

```
explain select candidates.* from (select * from objects_snapshot and owner_id = '\x34f483dfc35589a1b78052d1b34ee5c6d8168a2dd982c54edf8fac7a0943a255') candidates left join (select object_id, object_version from objects_history where checkpoint_sequence_number between 30000 and 40000) newer on (candidates.object_id = newer.object_id and candidates.object_version < newer.object_version) where newer.object_version is null  order by candidates.object_id asc limit 50;
```

In this query, the planner leverages indexes on `objects_snapshot` filtering, and then uses the `object_id_object_version_checkp_idx` on `objects_history`, so that's the one index we don't actually want to modify

Reply:
makes sense!

> I think we would benefit more from having the existing indexes replaced with partial indexes, except for the index on (object_id, object_version, checkpoint_sequence_number)

however to replace it with partial indices, looks like PG does not allow me to do that in place. I will hold the indices change for now and do that after pruning is enabled so that can do that on only the last 2 partitions.

Reply:
That's smart, but I worry that objects queries will degrade in performance in the meantime ... but I can approve this to unblock and get the objects snapshot changes rolling (for removing wrapped and deleted)

Reply:
Could we create the partial indexes additively, and then drop the old indexes when they're ready

Reply:
makes sense, added a migration for that -- the clock is ticking after merging tho for the new indices to be ready, otherwise they will run on next release, the droppings are quick, hopefully we can enable pruning soon and can only create new partial indices for the last 2 partiions.